indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
+                style2.alignment = TA_CENTER
+                caption = []
+                for session_id in self._showSessions:
+                    sess = self._conf.as_event.sessions.filter_by(id=session_id, is_deleted=False).first()
+                    if sess:
+                        caption.append(sess.title)
+                p = Paragraph("\n".join(caption), style2)
                 self._story.append(p)
-            self._story.append(Spacer(inch, 2*cm))
+            self._story.append(Spacer(inch, 2 * cm))
             self._story.append(self._toc)
             self._story.append(PageBreak())
 
-    def firstPage(self,c,doc):
+    def firstPage(self, c, doc):
         if self._ttPDFFormat.showCoverPage():
             c.saveState()
             if self._ttPDFFormat.showLogo():
                 self._drawLogo(c, False)
+
             height = self._drawWrappedString(c, self._conf.getTitle())
-            c.setFont('Times-Bold',modifiedFontSize(15, self._fontsize))
-            height-=2*cm
-            c.drawCentredString(self._PAGE_WIDTH/2.0,height,
-                    "%s - %s"%(self._conf.getAdjustedStartDate(self._tz).strftime("%A %d %B %Y"),
-                    self._conf.getAdjustedEndDate(self._tz).strftime("%A %d %B %Y")))
-            if self._conf.getLocation():
-                height-=2*cm
-                c.drawCentredString(self._PAGE_WIDTH/2.0,height,escape(self._conf.getLocation().getName()))
+            c.setFont('Times-Bold', modifiedFontSize(15, self._fontsize))
+            height -= 2 * cm
+            c.drawCentredString(self._PAGE_WIDTH / 2.0, height,
+                                "{} - {}".format(self._conf.getAdjustedStartDate(self._tz).strftime("%A %d %B %Y"),
+                                                 self._conf.getAdjustedEndDate(self._tz).strftime("%A %d %B %Y")))
+            if self._conf.as_event.location_data['venue_name']:
+                height -= 2 * cm
+                c.drawCentredString(self._PAGE_WIDTH / 2.0, height,
+                                    escape(self._conf.as_event.location_data['venue_name']))
             c.setFont('Times-Bold', modifiedFontSize(30, self._fontsize))
-            height-=1*cm
-            c.drawCentredString(self._PAGE_WIDTH/2.0,height,\
-                    self._title)
-            self._drawWrappedString(c, "%s / %s"%(self._conf.getTitle(),self._title), width=inch, height=0.75*inch, font='Times-Roman', size=modifiedFontSize(9, self._fontsize), color=(0.5,0.5,0.5), align="left", maximumWidth=self._PAGE_WIDTH-3.5*inch, measurement=inch, lineSpacing=0.15)
-            c.drawRightString(self._PAGE_WIDTH-inch,0.75*inch,
-                    nowutc().strftime("%A %d %B %Y"))
+            height -= 1 * cm
+            c.drawCentredString(self._PAGE_WIDTH / 2.0, height, self._title)
+            self._drawWrappedString(c, "{} / {}".format(self._conf.getTitle(), self._title), width=inch,
```

Comment:
`s/self._conf.getTitle()/self._conf.as_event.title/g`

Actually, if it's not too much work, we could even have a `self._event` var with the actual new-style event. It would make it easier to adapt in the future.